views.py

```diff
@@ -75,7 +80,7 @@ def download(request, path, rest_call=False, use_async=True, *args, **kwargs):
 
     bag_modified = "false"
     metadata_dirty = "false"
-    if istorage.exists(res_root):
+    if istorage.exists(res_root) and not is_zip_download:
```

Comment:
The condition check ```istorage.exists(res_root)``` is needed for zip download as well, so it is better to put ```if is_zip_download``` conditional block within this block, i.e., when ```istorage.exists(res_root)``` is true.

Reply:
That conditional block is specifically for the bags.  Bags persist until the resource changes and then is replaced.  Zips however are only created as needed and destroyed shortly after, so there is no reason to go into this block.  In an email exchange earlier with you, we talked about removing zips after they are downloaded.

Reply:
I know, but this condition check ```istorage.exists()``` is to check whether the resource exists or not, NOT for checking whether the temporary zip file exists or not. Note the input ```res_root``` which is the resource top directory. This is just a precaution to make sure resource actually exists since the hacker can pass in any random uuids which don't represent a resource stored in Hydroshare.

Reply:
Ok, so I can see argument for putting istorage.exists(res_root) on line 103  `if is_zip_download and istorage.exists(res_root):`

However, I don't see how this block stops someone from passing in any random uuid.  Both bag_modified and metadata_dirty are set in this block, but the code that checks those variables allow None value to pass through and create the bag.  The check for is_bag_download is not within this block and will always run.

Reply:
Agreed. Multiple people worked on this view function and this has not functioned as it should to capture random uuid being passed in. Now that you are at it, could you just modify the code to catch this upfront without creating the bags but instead just raise an exception or return a 400 bad request response?

Reply:
Did some rearranging following what we talked about with the largest change checking if the resource exists and returning a 400